test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlReaderTests.cs

```diff
@@ -289,6 +290,77 @@ public void ValidateNavigationPropertyBindingPathEndingInTypeCast(string odataVe
 
         }
 
+        [InlineData("4.0")]
+        [InlineData("4.01")]
+        [Theory]
+        public async Task ValidateNavigationPropertyBindingPathEndingInTypeCast_Async(string odataVersion)

```

Comment:
Can we move these async tests to a new class to avoid making this class so huge. We can have a new test class: `CsdlReaderAsyncTests`  The same for the CsdlWriterTests class. `CsdlWriterAsyncTests`

Reply:
@ElizabethOkerio I have moved all the async methods to individual `.Async partial classes`